### DISCUSSION TITLE
Pact provider state parameter support for pact js

### DISCUSSION THREAD

**Discussion by BhaskarPhukan** [pact-foundation/pact-js#698] [Q&A]
Hi , Is provider state parameter support for pact js  there ? if not , is there any alternative provided?

**Comment by mefellows** [comment]
Yes, please see the section on the home page for V3 pact support.

**Reply by BhaskarPhukan** [reply]
Thanks, let me have a look.

**Reply by BhaskarPhukan** [reply]
This looks like the parameter support for the provider support. How do i use these parameters on the customer contract tests?

**Reply by mefellows** [reply]
The `given` function has an optional `parameters` argument: https://github.com/pact-foundation/pact-js/blob/feat/v3.0.0/src/v3/pact.ts#L166-L169

I'm not sure what you mean by "on the customer contract tests"? I assume you mean "consumer", but also, why would you need them in the consumer tests? You control the test, including the parameters passed in to the `given` method, so have access to them already. Can you please explain your use case?
